docs/decisions/0019-forced-redemption.rst

```diff
@@ -0,0 +1,48 @@
+0019 Forced Redemption
+**********************
+
+Status
+======
+
+Proposed - March 2023
+
+Context
+=======
+There is frequently a need to force through a redemption (and related enrollment/fulfillment)
+of a particular learner, covered by a particular subsidy access policy, into
+some specific course run. This needs exists for reasons related to upstream
+business constraints, notably in cases where a course is included in a policy's catalog,
+but the desired course *run* is not discoverable due to the current state of its metadata.
+
+Some recent work has provided an ``allow_late_enrollment`` directive to be included
+in the ``metadata`` of the subsidy redeem payload.  This solution will utilize that directive,
+because it causes enrollment and/or external fulfillment to be forced through in
+downstream systems.
+
+Decision
+========
+* We'll expose a model via Django Admin that allows staff users to force redemption
+  for a given learner, into a particular course, under a particular subsidy access policy.
+* The model should describe which learner is being enrolled, via which policy, and into
+  which specific course **run**.  It should also record which staff user wrote the record
+  when it was created/modified, and when the redemption succeeded or failed.
+  Furthermore, it should store a reference to the created
+  transaction on success, and store a reference to any error information on failure.
+* It should be possible to write a record for this model *without* the forced redemption immediately
+  taking place, although perhaps the default behavior should cause the redemption to
+  occur on save.
+* All the standard redeemability logic will still hold, that is, ``can_redeem()`` must return ``True``
+  for the redemption to take place.  This entails the automated creation of a ``LearnerContentAssignment``
+  record for assignment-based policies.
+
+Consequences
+============
```

Comment:
Should this clarify that the tool will not work to force Exec Ed redemptions due to the T's&C's flow requiring learner intervention?

Reply:
Or is there more to this plan specifically around Exec Ed courses not yet written?  E.g. I can imagine the new force redemption model could indicate intent to force a specific learner-policy-run redemption, then an accompanying UI component treats it like an assignment (even if the policy type is auto-apply).  But this is completely made up and I'm just using my imagination to fill in the gaps.

Reply:
Updated to reflect that this won't work for EE content.